packages/layout/src/tree-layout/layout-definition/helper/index.js

```diff
@@ -0,0 +1,290 @@
+    if (typeof value !== 'string' || typeof compareTo !== 'string') {
+        throw new TypeError('value and compareTo must be string');
+    }
+    return value.toLowerCase() === compareTo.toLowerCase();
+}
+
+export const getNodeId = (() => {
+    let _uid = 0;
+    return () => `${NODE_PREFIX}-${++_uid}`;
+})();
+
+export function yExtraSpace (node) {
+    let smallestHeight = 0;
+    if (node.getCutType() === 'v') {
+        smallestHeight = smallestExtraHeightHorizontally(node); // eslint-disable-line no-use-before-define
+    } else if (node.getCutType() === 'h') {
+        node.children().forEach((child) => {
+            smallestHeight += yExtraSpace(child);
+        });
+    } else if (node.model().host() && node.model().host().getLogicalSpace) {
+        const containerHeight = node.boundBox().height;
+        const hostHeight = node.model().host().getLogicalSpace().height;
+
+        smallestHeight = containerHeight - hostHeight;
+        if (smallestHeight < 0) {
+            smallestHeight = 0;
+        }
+    } else {
+        smallestHeight = 0;
+    }
+    return smallestHeight;
+}
+
+export function smallestExtraHeightHorizontally (node) {
+    let smallestHeight = Number.MAX_SAFE_INTEGER;
+    node.children().forEach((child) => {
+        const h = yExtraSpace(child);
+        if (h < smallestHeight) {
+            smallestHeight = h;
+        }
+    });
+    return smallestHeight;
+}
+
+export function xExtraSpace (node) {
+    let smallestWidth = 0;
+    if (node.getCutType() === 'h') {
+        smallestWidth = smallestExtraWidthVertically(node); // eslint-disable-line no-use-before-define
+    } else if (node.getCutType() === 'v') {
+        node.children().forEach((child) => {
+            smallestWidth += xExtraSpace(child);
+        });
+    } else if (node.model().host() && node.model().host().getLogicalSpace) {
+        const containerWidth = node.boundBox().width;
+        const hostWidth = node.model().host().getLogicalSpace().width;
+        smallestWidth = containerWidth - hostWidth;
+        if (smallestWidth < 0) {
+            smallestWidth = 0;
+        }
+    } else {
+        smallestWidth = 0;
+    }
+    return smallestWidth;
+}
+
+export function smallestExtraWidthVertically (node) {
+    let smallestWidth = Number.MAX_SAFE_INTEGER;
+    node.children().forEach((child) => {
+        const w = xExtraSpace(child);
+        if (w < smallestWidth) {
+            smallestWidth = w;
+        }
+    });
+    return smallestWidth;
+}
+
+export function determineBoundBox (bb, i, arr, instance) {
+    if (i) {
+    // if not first sibling, take boundbox from previous sibling
+        const lastSibling = arr[i - 1];
+        return {
+            width: bb.width,
+            height: bb.height,
+
+            top: instance.parentCut() === 'h'
+        ? lastSibling.boundBox().top + lastSibling.boundBox().height : lastSibling.boundBox().top,
+
+            left: instance.parentCut() === 'h'
```

Comment:
Use strategy patter for similar conditions if cant be clubbed together.
Multiple if else statements or conditional statements reduce the scalability of the code